romancal/source_catalog/source_catalog_step.py

```diff
@@ -95,39 +95,29 @@ def process(self, step_input):
             model.err = input_model.err.copy()
             model.weight = input_model.weight
 
-        if isinstance(model, ImageModel):
-            cat_model = datamodels.ImageSourceCatalogModel
-        else:
-            cat_model = datamodels.MosaicSourceCatalogModel
-        source_catalog_model = cat_model()
-
-        source_catalog_model.meta = {}
-        for key in source_catalog_model.meta._schema_attributes.explicit_properties:
-            value = (
-                model.meta.instrument[key]
-                if key == "optical_element"
-                else model.meta[key]
-            )
-            source_catalog_model.meta[key] = value
-
-        if self.forced_segmentation:
-            source_catalog_model.meta["forced_segmentation"] = self.forced_segmentation
-            forced = True
-        else:
-            forced = False
+        # return empty segmentation image and catalog if all pixels are
+        # masked
+        if np.all(mask):
+            self.log.warning("Cannot create source catalog. All pixels are masked.")
+            segment_img = np.zeros(model.data.shape, dtype=np.uint32)
+            cat = Table()
```

Comment:
@larrybradley thanks for your patience with this. There is new API for the various source catalog models that should allow this to be replaced with a call to `create_empty_catalog`:
https://github.com/spacetelescope/romancal/blob/803b2fe7388108cec7f6f9d835f208f55c2b1a77/romancal/source_catalog/source_catalog_step.py#L156
to generate a schema-valid catalog with no rows.

Reply:
Thanks, @braingram.  I've updated this PR.